Fix various problems - main one was systematic execution of .py scripts, regardles of md5 hash check

mchaaler opened the pull request:
Since I'm not confident enough with tests, I did not add the corresponding non-regression tests. Eager to learn by looking at how you'll implemented them!

smarie commented:
Thanks a lot @mchaaler ! Very nice

I see three things in this PR:

 - developer doc fix and git ignore update
 - the extra `.resolve()` statements : are these related to a specific platform or use case ? what was not working before this ? 
 - the thumbnail resolution when the script is not re-run (`[gen_single.py](https://github.com/smarie/mkdocs-gallery/pull/27/files#diff-6c01323da63ab6f1db29efb3284b8ce526bdbe9da4e622f5b19e7d71b090c994)`)

Can you please create one issue for each of these bullet points, so that we can relate the PR ? We can keep this single PR fixing the 3 at once. 

In particular I do not understand how the thumbnail resolution fix I see in the PR, relates to the title in this PR (python scripts being re-run even if the md5 hash is the same). Can you please elaborate ? Thanks in advance !

mchaaler commented:
Hi!
Just created issues #28 & #29 corresponding to 1st and 3rd bullet points. Hope the comments help clarifying things a bit, by the way.

Considering the extra resolve() statements added in https://github.com/smarie/mkdocs-gallery/pull/27/commits/48966764b7a43c00908bf93aeb5fe88352390adc, I need to dig further. I am not able to reproduce the bug at home so far (worked on it from the office), although the platforms are similar (Windows 10/11, PyCharm & Git-bash or fish, conda virtual environment). The use cases are different though: working on mkdocs-gallery itself at home, on a business project at work.

mchaaler commented:
I think I got it.
Since I wanted to disable the "edit URL" link, as MkDocs documentation explains, I tried to set the "edit_uri" parameter to an empty string in mkdocs.yml.
This did not work because of issue #32.
I then decided (don't know exactly why...) to replace the empty string by "." (or "./"), which caused the problem mentioned in https://github.com/smarie/mkdocs-gallery/commit/48966764b7a43c00908bf93aeb5fe88352390adc, but actually did not give the expected result removing the "edit URL" link...

Just added a commit to the PR that fixes issue #32, and reverted https://github.com/smarie/mkdocs-gallery/commit/48966764b7a43c00908bf93aeb5fe88352390adc.

smarie commented:
All set, thanks @mchaaler !
